### PR TITLE
Changes to iris search processing algorithm.

### PR DESCRIPTION
**Pull request by anuran-roy**
Here, I updated the search results for the site processing algorithm in iris. Can you all please check it?

### CODE REVIEW

**Comment by anuran-roy** [comment]
Update

**Comment by anuran-roy** [comment]
Update on `iris > meta_search_util.py`

File: core/util/iris/meta_search_util.py [hunk 1]
```diff
@@ -1,53 +1,142 @@
...
-		path = "/".join(path_join)
-		path = path.replace('/', ' › ')
-		cite = f"{host} › {path}"
-		return cite
+		self.framework=main.framework
+     
+	def make_cite(self, url: 'URL String') -> 'cite':
+		default_pages=['index']
+		details=self.framework.urlib(url)
+		qs=details[4]
```
Comment (saeeddhqan) [inline core/util/iris/meta_search_util.py]: something is wrong with this code. details var is not subscribable.

**Review by saeeddhqan — COMMENTED** [review]
details is not subscriptable.
e

**Comment by anuran-roy** [comment]
@saeeddhqan can you try it now?

**Comment by saeeddhqan** [comment]
the first result is the origin and the second is for this pr:
ee
As you can see, it has some bugs.
also, the origin had a bug that I fixed:
eee

**Comment by anuran-roy** [comment]
I actually think that although there is no hard and fast rule, there is also few or no site names with such long domain names... Except for onion sites... All sites are recommended to use around 75 character long URLs, and for deeply linked sites, around 100 is the recommended maximum. 😅😅

**Comment by saeeddhqan** [comment]
Yes, I should fix it.

**Comment by anuran-roy** [comment]
> Yes, I should fix it.

No I meant, for all practical purposes, my algo should be ok. Few (if not no people) use super-large domain names. Atleast that's what I guess. 😅

**Comment by anuran-roy** [comment]
> Yes, I should fix it.

No I meant, for all practical purposes, my algo should be ok. Few (if not no people) use super-large domain names. Atleast that's what I guess. :sweat_smile

**Comment by saeeddhqan** [comment]
No, it's not okay. look at the pictures.

**Comment by anuran-roy** [comment]
Oh okay... I got it. The final path. Yeah I'll surely work on it and update you 😃

**Comment by saeeddhqan** [comment]
we don't need to show parameters and comments. so, remove those. also if we enter something like this: https://google.com/ we got: https://google.com > ...
even though we don't have any rest!

**Comment by anuran-roy** [comment]
Oh ok. I am working on that. I developed another script. The new one is just around 20 lines long. 

EDIT: I looked up online as to how google processes long domain names on the results it shows. I found that we were kinda overcomplicating it. 

Screenshot: 

1. 
image

(From mobile)

2.
image

(From PC)

As we can see, it is rather simple, picking up 34 characters for mobile search, and the whole domain name in case of PC search.

**Comment by saeeddhqan** [comment]
well, 110 is too long, and also google doesn't show ?query and comments btw. also it shows 110 char like this ... > which is not good.

**Comment by anuran-roy** [comment]
Uhh, so I found what the length of the longest domain in the world is. Since we need atleast the domain name to come properly, I chose that length as the threshold. Turns out it is 85 characters long. So can we change the character limit from 110 to 85?

**Comment by saeeddhqan** [comment]
You haven't used the default page. and also google doesn't show suffixes but it does. something like this? `https://google.com > index.php` or it may show something like this: `https://google.com > xx[to 85 char] > fox.htm...`

**Pull request closed without merging**